app/controllers/push_notifications_controller.rb

```diff
@@ -0,0 +1,64 @@
+require_dependency 'push_notifications'
+
+class PushNotificationsController < ApplicationController
+  layout false
+  before_filter :ensure_logged_in
+  before_filter :set_endpoint, only: [:subscribe, :unsubscribe]
+  skip_before_filter :preload_json
+
+  def subscribe
+    if @endpoint.start_with?(PushNotifications::GCM_ENDPOINT)
+      PushNotifications.subscribe(current_user, @endpoint)
+      render json: success_json
+    else
+      render json: failed_json
+    end
+  end
+
+  def unsubscribe
+    if @endpoint.start_with?(PushNotifications::GCM_ENDPOINT)
+      PushNotifications.unsubscribe(current_user, @endpoint)
+      render json: success_json
+    else
+      render json: failed_json
+    end
+  end
+
+  def latest
+    guardian.ensure_can_see_notifications!(current_user)
+
+    notifications = Notification.where(user_id: current_user.id, read: false)
+                                .visible
+                                .includes(:topic)
+                                .order(created_at: :desc)
+
+    base_url = Discourse.base_url
+    site_title = SiteSetting.title
+
+    info =
+      if notifications.count == 1
```

Comment:
If you invert this conditional you can use `.many?`
http://api.rubyonrails.org/classes/Enumerable.html#method-i-many-3F